graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40055 graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category**
* #39413 graph mode: util for fusion of functions which require observation
* #39203 quant layer/group/instance norm: make weights and biases optional

Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

vkuzo committed:
graph mode: move hardsigmoid op to `single_input_general_value` category

Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: move hardsigmoid op to `single_input_general_value` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization/helper.cpp:
```diff
@@ -137,6 +137,8 @@ std::vector<std::string> _single_input_general_value_aten_funcs = {
     "elu_",
     "leaky_relu",
     "leaky_relu_",
+    "hardsigmoid",
+    "hardsigmoid_",
```
this is fixed qparams op, see _fixed_qparams_map, you will need to include this in `getPassThroughInputs` function

vkuzo replied:
from looking at `getPassThroughInputs`, seems like this diff would not change the behavior?  I might be misunderstanding your suggestion.

`hardsigmoid` has fixed output qparams and it does change the input values, which is why it seemed it belongs in `_single_input_general_value_aten_funcs`.  I could be missing something.

jerryzh168 replied:
we will need to include a check for fixed input qparam ops, by calling `getFixedQParams` in `getPassThroughInputs`

`hardsigmoid` has fixed qparams, we have a separate category for that, which is defined in `_fixed_qparams_map`

vkuzo replied:
I see, looks like there is an assumption in the current code that the ops with fixed params are not included in things like `_single_input_general_shape_call_funcs`.

I'm going to leave this diff at just the bug fix (since hardsigmoid seems to be left over in the wrong category), and we can do the refactor if needed in a separate PR.

vkuzo committed:
Update on "graph mode: move hardsigmoid op to `single_input_general_value` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: move hardsigmoid op to `single_input_general_value` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]

jerryzh168 reviewed (approved):
actually it's included in https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/jit/passes/quantization/helper.cpp.html#_ZN5torch3jit32isSingleInputGeneralAtenFunctionEPNS0_4NodeE, so no further fix is needed.

vkuzo committed:
Update on "graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]

vkuzo committed:
Update on "graph mode: remove hardsigmoid op from `_single_input_general_shape_call_funcs` category"


Summary:

Noticed this while reading the `helper.cpp` file, seems like this op
should be in the `single_input_general_value` bucket.

Test Plan:

```
python test/test_quantization.py TestQuantizeScriptPTSQOps
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22054257](https://our.internmc.facebook.com/intern/diff/D22054257)

[ghstack-poisoned]
